include/blmc_drivers/blmc_joint_module.hpp

```diff
@@ -695,6 +709,68 @@ class BlmcJointModules
         return homing_status;
     }
 
+    /**
+     * @brief Perform homing for all joints at endstops.
+     *
+     * If one of the joints fails, the complete homing fails.  Otherwise it
+     * loops until all joints finished.
+     * If a joint is finished while others are still running, it is held at the
+     * home position.
+     *
+     * See BlmcJointModule::update_homing for details on the homing procedure.
+     *
+     * See BlmcJointModule::init_homing_at_endstop for description of the arguments.
+     *
+     * @return Final status of the homing procedure (either SUCCESS if all
+     *     joints succeeded or the return code of the first joint that failed).
+     */
+    HomingReturnCode execute_homing_at_endstop(
+        Vector home_offset_rad)
+    {
+        // Initialise homing for all joints
+        for (size_t i = 0; i < COUNT; i++)
+        {
+            modules_[i]->init_homing_at_endstop((int)i,
+                                                home_offset_rad[i]);
+        }
+
+        return check_homing_status();
```

Comment:
I might be missing something, but is it actually necessary to call `check_homing_status()` here as this already succeeds directly in the init step?

Reply:
Yes, you are right. I removed all the state machine stuff in case of homing_with_index = False